File: rakugan/demo/about.pl
# Links that open a page, and the system clipboard.
use Rakugan;

class About {
    use Rakugan;
    field $status = "";

    method view {
        return column(
            text("Rakugan", size => 28),
            text("version 0.1.0"),
            link("Website", "https://i2y.github.io/yokan/"),
            link("Source", "https://github.com/i2y/yokan"),
            link("Docs", "https://i2y.github.io/yokan/tour/"),
            button("copy link", on_click => sub {
                clipboard_set_text("https://github.com/i2y/yokan");
                $status = "copied";
            }),
            text("status: $status"),
            spacing => 8,
            padding => 14,
        );
    }
}

run(About->new, title => "about");

File: gomamochi/demo/about.go
// Links that open a page, and the system clipboard.
package main

import . "github.com/i2y/yokan/gomamochi"

type About struct {
	status string
}

func (a *About) View() Element {
	return Column(
		Text("Gomamochi").Size(28),
		Text("version 0.1.0"),
		Link("Website", "https://i2y.github.io/yokan/"),
		Link("Source", "https://github.com/i2y/yokan"),
		Link("Docs", "https://i2y.github.io/yokan/tour/"),
		Button("copy link").OnClick(func() {
			ClipboardSetText("https://github.com/i2y/yokan")
			a.status = "copied"
		}),
		Text("status: "+a.status),
	).Spacing(8).Padding(14)
}

func main() {
	Run(&About{}, Title("about"))
}
